pom.xml

```diff
@@ -1606,6 +1614,16 @@
                 <artifactId>commons-io</artifactId>
                 <version>${commons-io.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.apache.commons</groupId>
+                <artifactId>commons-text</artifactId>
+                <version>${commons-text.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.github.javaparser</groupId>
+                <artifactId>javaparser-core</artifactId>
+                <version>${javaparser.version}</version>
+            </dependency>
```

Comment:
Any new dependency has to be productized before being merged. The higher the number of (direct or indirect) dependencies, the higher cost due to maintenance of CVE fixes.

Can you provide the list of all the dependencies brought in via this dependency?

----

For the reference, link to the previous discussion: https://github.com/keycloak/keycloak/pull/11806/files/0a3f7a57b23af4e605d17dec8206c726529336d7#diff-ff141e665c6e337f689690d7372e2f412975ca68332ddeaae08d0425cc0cfa83

Reply:
Thanks for weighing in @hmlnarik !
`javaparser-core` is self-contained and doesn't have transitive dependencies:
https://mvnrepository.com/artifact/com.github.javaparser/javaparser-core/3.24.2

We are, anyhow, using it in the operator, transitively brought by `sundrio` wich is [directly used](https://github.com/keycloak/keycloak/blob/3ff3aeba29bc4d66d30a7d6243db8ca42ccaf3fa/operator/src/main/java/org/keycloak/operator/v2alpha1/crds/Keycloak.java#L33) and [in-directly used](https://github.com/keycloak/keycloak/blob/3ff3aeba29bc4d66d30a7d6243db8ca42ccaf3fa/operator/pom.xml#L58-L67):
https://github.com/sundrio/sundrio/blob/72cb33cebe426152fa863b4623884e8760de372d/pom.xml#L212-L216

If I understand the process correctly we would need to productize it anyhow. Would you mind helping to understand the next steps? 🙏

Reply:
> `javaparser-core` is self-contained and doesn't have transitive dependencies:

Sadly that is not true, there are number of custom plugins: https://repo1.maven.org/maven2/com/github/javaparser/javaparser-core/3.24.2/javaparser-core-3.24.2.pom

All of them need to be productized, and any CVE in them watched for.

> We are, anyhow, using it in the operator, transitively brought by `sundrio` wich is [directly used](https://github.com/keycloak/keycloak/blob/3ff3aeba29bc4d66d30a7d6243db8ca42ccaf3fa/operator/src/main/java/org/keycloak/operator/v2alpha1/crds/Keycloak.java#L33) and [in-directly used](https://github.com/keycloak/keycloak/blob/3ff3aeba29bc4d66d30a7d6243db8ca42ccaf3fa/operator/pom.xml#L58-L67):

Good catch. Who is now responsible for maintaining it?

Cc: @stianst @mposolda @pedroigor

Reply:
An additional consideration is that `javaparser-core` is used in a core component of Quarkus:
https://github.com/quarkusio/quarkus/blob/748e13113ff0f57fdfa6de669eecb5f44116a2a6/core/processor/pom.xml#L27

Reply:
This discussion moved to internal since it's relevant for the commercial offering. I'll re-open it if necessary 👍  thanks for the contributions!